proof/append.go

```diff
@@ -160,6 +160,46 @@ func CreateTransitionProof(prevOut wire.OutPoint,
 		TapSiblingPreimage: params.TapscriptSibling,
 	}
 
+	if proof.Asset.IsTransferRoot() {
+		stxoInclusionProofs := make(
+			map[asset.SerializedKey]commitment.Proof,
+			len(proof.Asset.PrevWitnesses),
+		)
+		for _, wit := range proof.Asset.PrevWitnesses {
+			if wit.PrevID == nil {
+				return nil, fmt.Errorf("witness %v has no "+
+					"prevID", wit)
+			}
+
+			spentAsset := &asset.Asset{
+				ScriptKey: asset.NewScriptKey(
+					asset.DeriveBurnKey(*wit.PrevID),
+				),
+			}
+
+			// Generate an STXO inclusion proof for each prev
+			// witness.
+			_, stxoProof, err := params.TaprootAssetRoot.Proof(
+				asset.EmptyGenesisID,
+				spentAsset.AssetCommitmentKey(),
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			key := spentAsset.ScriptKey.PubKey.SerializeCompressed()
```

Comment:
If the code is moved from the last commit here, then this can just be: `keySerialized := asset.ToSerialized(scriptKey.PubKey)`